tools/clang/lib/CodeGen/CGHLSLMS.cpp

```diff
@@ -965,8 +974,12 @@ unsigned CGMSHLSLRuntime::ConstructStructAnnotation(DxilStructAnnotation *annota
 
     fieldAnnotation.SetCBufferOffset(CBufferOffset);
     fieldAnnotation.SetFieldName(fieldDecl->getName());
-    if (!fieldSemName.empty())
+    if (!fieldSemName.empty()) {
       fieldAnnotation.SetSemanticString(fieldSemName);
+
+      if (m_PreciseOutputSet.count(fieldSemName))
+        fieldAnnotation.SetPrecise();
```

Comment:
So all structs with this semantic name will get precise set on the field?  What happened to only impacting the entry function?  This would impact any case where this field in this structure is used, even if not used specifically to output position.  This might not hurt much, and may be more rare, but probably not "precisely" what you want.  A more limited effect would be more difficult of course.